classes/Hook.php

```diff
@@ -1244,4 +1253,17 @@ private static function getMethodName(string $hookName): string
     {
         return 'hook' . ucfirst($hookName);
     }
+
+    /**
+     * Return an instance of the specified hook.
+     *
+     * @param string $hook_name Hook name
+     *
+     * @return Hook|false
+     */
+    public static function getInstanceByName($hook_name)
+    {
+        $hook = new Hook((int) Hook::getIdByName($hook_name));
```

Comment:
You should create a custom query, here you're using a query in `getIdByName`, then another query to retrieve the data from an id.

Reply:
Exact ! On the other hand, travisCI seems broken ... Can you look please ?